javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/OpenTelemetryInstaller.java

```diff
@@ -44,22 +49,49 @@ public static AutoConfiguredOpenTelemetrySdk installOpenTelemetrySdk(
       return SdkAutoconfigureAccess.create(
           sdk,
           SdkAutoconfigureAccess.getResource(autoConfiguredSdk),
-          new DeclarativeConfigPropertiesBridgeBuilder()
-              .addMapping("otel.javaagent", "agent")
-              // these properties are used to initialize the SDK before the configuration file
-              // is loaded for consistency, we pass them to the bridge, so that they can be read
-              // later with the same value from the {@link DeclarativeConfigPropertiesBridge}
-              .addOverride(
-                  "otel.javaagent.debug", earlyConfig.getBoolean("otel.javaagent.debug", false))
-              .addOverride(
-                  "otel.javaagent.logging", earlyConfig.getString("otel.javaagent.logging"))
-              .buildFromInstrumentationConfig(configProvider.getInstrumentationConfig()),
+          getDeclarativeConfigBridgedProperties(earlyConfig, configProvider),
           configProvider);
     }
 
     return autoConfiguredSdk;
   }
 
+  // Visible for testing
+  static ConfigProperties getDeclarativeConfigBridgedProperties(
+      EarlyInitAgentConfig earlyConfig, ConfigProvider configProvider) {
+    return new DeclarativeConfigPropertiesBridgeBuilder()
+        .addMapping("otel.javaagent", "agent")
+        .addOverride("otel.instrumentation.common.default-enabled", defaultEnabled(configProvider))
+        // these properties are used to initialize the SDK before the configuration file
+        // is loaded for consistency, we pass them to the bridge, so that they can be read
+        // later with the same value from the {@link DeclarativeConfigPropertiesBridge}
+        .addOverride("otel.javaagent.debug", earlyConfig.getBoolean("otel.javaagent.debug", false))
+        .addOverride("otel.javaagent.logging", earlyConfig.getString("otel.javaagent.logging"))
+        .buildFromInstrumentationConfig(configProvider.getInstrumentationConfig());
+  }
+
+  private static boolean defaultEnabled(ConfigProvider configProvider) {
+    DeclarativeConfigProperties instrumentationConfig = configProvider.getInstrumentationConfig();
+    if (instrumentationConfig == null) {
+      return true;
+    }
+
+    String profile =
+        instrumentationConfig
+            .getStructured("java", empty())
+            .getStructured("agent", empty())
+            .getString("instrumentation_profile", "default");
```

Comment:
I know this is what we discussed in the SIG, and I do think it works, but I wonder if the word "profile" might be confusing and make people think this has something to do with profiling. If so, another possibility could be to use `instrumentation_mode`.  Maybe the fact that we can provide a template for configs for users and add a comment above this option that explains what it's for can be enough to disambiguate the profiling piece.